scripts/sct_straighten_spinalcord.py

```diff
@@ -455,15 +466,15 @@ def straighten(self):
                     if warp_space_y[0] < 0:
                         warp_space_y[1] += warp_space_y[0] - 2
                         warp_space_y[0] = 0
-                    if self.resample_factor != 0.0:
-                        sct.run(['sct_crop_image', '-i', 'tmp.centerline_pad_native.nii.gz', '-o', 'tmp.centerline_pad_crop_native.nii.gz',
+
+                    sct.run(['sct_crop_image', '-i', 'tmp.centerline_pad_native.nii.gz', '-o', 'tmp.centerline_pad_crop_native.nii.gz',
                          '-dim', '0,1,2',
                          '-start', str(warp_space_x[0]) + ',' + str(warp_space_y[0]) + ',0',
                          '-end', str(warp_space_x[1]) + ',' + str(warp_space_y[1]) + ',' + str(end_point_coord_native[2] - start_point_coord_native[2]),
                         ])
 
                     fname_ref = 'tmp.centerline_pad_crop_native.nii.gz'
-                    xy_space = 40
+                    self.xy_size = 40
```

Comment:
now that `xy_size` was made a class member, its changing throughout the code looks weirder.
It's set to 35 in the constructor, then reset at 35 in some cases, or modified to 40 in some cases.

Reply:
You are right. The change of value for xy_size was not necessary. I removed it.